service/src/main/java/greencity/service/google/GoogleApiService.java

```diff
@@ -68,4 +75,65 @@ public GeocodingResult getGeocodingResultByCityAndCountryAndLocale(String countr
             throw new GoogleApiException(e.getMessage());
         }
     }
+
+    /**
+     * Retrieves the address details from Google API based on the given geographical
+     * coordinates. This method sends a request to the Google Geocoding API with the
+     * provided latitude and longitude. It processes the response to extract address
+     * components such as city and region. If the API call fails or no results are
+     * found, it returns {@code null}.
+     *
+     * @param coordinates the {@link LatLng} object containing the latitude and
+     *                    longitude.
+     * @return an {@link AddressResponseFromGoogleAPI} containing city, region, and
+     *         district details, or {@code null} if the API call fails or no results
+     *         are returned.
+     */
+    public AddressResponseFromGoogleAPI getResultFromGoogleByCoordinates(LatLng coordinates) {
+        try {
+            GeocodingResult[] results = GeocodingApi.newRequest(context)
+                .latlng(coordinates)
+                .language(UKRAINIAN.getLanguage())
+                .await();
+
+            return Optional.ofNullable(results)
+                .filter(r -> r.length > 0)
+                .map(r -> getAddressResponse(r[0].addressComponents))
+                .orElse(null);
+        } catch (IOException | InterruptedException | ApiException e) {
+            log.error("Error during the call to Google API, reason: {}", e.getMessage());
+            Thread.currentThread().interrupt();
```

Comment:
Why do we need  `Thread.currentThread().interrupt();`?

Reply:
If an InterruptedException occurs, it indicates that someone interrupted the thread, for example to stop execution. Calling Thread.currentThread().interrupt() restores this interrupt status so that other parts of the program can respond appropriately to the interruption. I will push changes little later.

catch (IOException | ApiException e) {
            log.error("Error during the call to Google API, reason: {}", e.getMessage());
            return null;
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            return null;
        }